modules/navier_stokes/src/fvkernels/INSFVMomentumAdvection.C

```diff
@@ -59,15 +57,16 @@ INSFVMomentumAdvection::validParams()
   // We need 2 ghost layers for the Rhie-Chow interpolation
   params.set<unsigned short>("ghost_layers") = 2;
 
+  params.addParam<UserObjectName>("velocity_interpolator", "Velocity interpolation user object");
```

Comment:
Should we just make this required? How is interpolation even going to work without it since you removed all the class data members?

Reply:
Average interpolation would just use the FVUtil

Reply:
See the PR first post, I on the fence about creating the same object for average interpolation

Reply:
I think a base `FVInterpolator` makes sense. IMO that object should not just be dedicated to average interpolation. Instead I think the base should take an enum parameter that would allow it to do any of the interpolations introduced with the `Limiter` classes in the CNSFV PR